src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_WellKnownAttributes.cs

```diff
@@ -8878,5 +8878,50 @@ public static void Main()
                 // using static BothObsoleteParent.BothObsoleteChild;
                 Diagnostic(ErrorCode.WRN_DeprecatedSymbol, "BothObsoleteParent.BothObsoleteChild").WithArguments("BothObsoleteParent.BothObsoleteChild").WithLocation(5, 14));
         }
+
+        [Fact, WorkItem(19394, "https://github.com/dotnet/roslyn/issues/19394")]
+        public void WellKnownTypeAsStruct_DefaultConstructor_DynamicAttribute()
+        {
+            var code = @"
+namespace System.Runtime.CompilerServices
+{
+    public struct DynamicAttribute
+    {
+        public DynamicAttribute(bool[] transformFlags)
+        {
+        }
+    }
+}
+class T
+{
+    void M(dynamic x) {}
```

Comment:
>dynamic [](start = 11, length = 7)

What is the behavior for ```dynamic[]``` type? Should we disallow structures in ICustomAttribute altogether regardless of what constructor is used?

Reply:
The crash only happens with default constructors. For structs and non default constructors, we emit successfully. Changing that would be a breaking change that we need to take to LDM.

---
In reply to: [188722425](https://github.com/dotnet/roslyn/pull/26815#discussion_r188722425) [](ancestors = 188722425)

Reply:
> The crash only happens with default constructors. For structs and non default constructors, we emit successfully. Changing that would be a breaking change that we need to take to LDM.

Could you please open an issue to follow-up or start a conversation with Compatibility Council? This is not blocking, but it would be good to make a decision on this sooner rather than later. 

---
In reply to: [189076550](https://github.com/dotnet/roslyn/pull/26815#discussion_r189076550) [](ancestors = 189076550,188722425)